scripts/train_recognizer_synthtext90_multigpu.py

```diff
@@ -129,13 +129,14 @@ def get_image_generator(filepaths, augmenter, width, height):
         ), math.ceil(len(filepaths) / args.batch_size))
         for filepaths, augmenter in [(training_filepaths, augmenter), (validation_filepaths, None)]
     ]
+    SynthText90_MAX_STRING_LENGTH = 48
     training_generator, validation_generator = [
         tf.data.Dataset.from_generator(
             functools.partial(recognizer.get_batch_generator,
                               image_generator=image_generator,
                               batch_size=args.batch_size),
             output_types=((tf.float32, tf.int64, tf.float64, tf.int64), tf.float64),
-            output_shapes=((tf.TensorShape([None, 31, 200, 1]), tf.TensorShape([None, 48]),
+            output_shapes=((tf.TensorShape([None, 31, 200, 1]), tf.TensorShape([None, SynthText90_MAX_STRING_LENGTH]), 
```

Comment:
Agreed that we should set this value with an explanation. The maximum string length is determined from the recognizer model build parameters. The longer the input width, the longer the maximum string length. The pooling size also has an impact but we hardly ever change that parameter. One can infer what the maximum string length is from the built model. See the following line for an example.

https://github.com/faustomorales/keras-ocr/blob/5e075d48584df8c07442ed618ebac2c7cfb10d8d/keras_ocr/recognition.py#L354

So I would ask that you `SynthText90_MAX_STRING_LENGTH` with `recognizer.training_model.input_shape[1][1] ` and then we can merge this in. Thanks!

Reply:
Ah, I see, thanks. Does that mean synthText90 images have 48 pixels for their width?

Reply:
No, SynthText90 images can be of any size and the word boxes within them can be of any size. 48 here refers to the maximum *number of characters* the recognizer model can predict, which is always equal to `(recognizer_input_width / 4) - 2`. Here, the `recognizer_input_width` is 200.

The reason we have to provide the maximum string length at all is because `tf.Dataset` requires that we specify what the shapes of generated batches will be. So, the maximum string length, 48, is *not* a property of the dataset but rather a property of the model which we can change by expanding the input width of images provided to the model.

Separately, because we use this value only once, I do not think we should  create a named variable. The current name, `SynthText90_MAX_STRING_LENGTH` is incorrect anyway. If anything, the variable would have been named something like `Recognizer_MAX_STRING_LENGTH`. So please change `tf.TensorShape([None, SynthText90_MAX_STRING_LENGTH])` to `tf.TensorShape([None, recognizer.training_model.input_shape[1][1]])`.

Reply:
You right, make a lot more sense. thanks. Also just a side question: Let's say I want to use the dataset generator for the recognizer using image generator, but I want to force the image generator to use all the text given by the text generator as a single line and not decompose it as many lines. What would be the easiest way, please Thanks

Reply:
If all you want is to obtain crops of single lines, that is handled by `keras_ocr.data_generation.convert_image_generator_to_recognizer_input`. If you want a full image but just want to avoid line breaks, you will have to use a combination of text generator and arguments to `get_image_generator` such that all the text always fits on a single line. Alternatively, you could use `draw_text_image` directly and thereby ensure that only a single line is used each time.

Reply:
Actually , the problem is that with `keras_ocr.data_generation.convert_image_generator_to_recognizer_input` I get a crop but the line is usually broken into multiple lines and I don't want the whole image just a crop containing all the text that was generated with the text generator. Let me try the two last suggestions thanks.